lightning/src/chain/channelmonitor.rs

```diff
@@ -1639,24 +1763,24 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 						if let Some(ref outpoints) = self.counterparty_claimable_outpoints.get($txid) {
 							for &(ref htlc, ref source_option) in outpoints.iter() {
 								if let &Some(ref source) = source_option {
-									log_info!(logger, "Failing HTLC with payment_hash {} from {} counterparty commitment tx due to broadcast of revoked counterparty commitment transaction, waiting for confirmation (at height {})", log_bytes!(htlc.payment_hash.0), $commitment_tx, height + ANTI_REORG_DELAY - 1);
-									match self.onchain_events_waiting_threshold_conf.entry(height + ANTI_REORG_DELAY - 1) {
-										hash_map::Entry::Occupied(mut entry) => {
-											let e = entry.get_mut();
-											e.retain(|ref event| {
-												match **event {
-													OnchainEvent::HTLCUpdate { ref htlc_update } => {
-														return htlc_update.0 != **source
-													},
-													_ => true
-												}
-											});
-											e.push(OnchainEvent::HTLCUpdate { htlc_update: ((**source).clone(), htlc.payment_hash.clone())});
-										}
-										hash_map::Entry::Vacant(entry) => {
-											entry.insert(vec![OnchainEvent::HTLCUpdate { htlc_update: ((**source).clone(), htlc.payment_hash.clone())}]);
+									self.onchain_events_waiting_threshold_conf.retain(|ref entry| {
+										if entry.height != height { return true; }
+										match entry.event {
+											 OnchainEvent::HTLCUpdate { ref htlc_update } => {
+												 htlc_update.0 != **source
+											 },
+											 _ => true,
 										}
-									}
+									});
+									let entry = OnchainEventEntry {
+										txid: *$txid,
+										height,
+										event: OnchainEvent::HTLCUpdate {
+											htlc_update: ((**source).clone(), htlc.payment_hash.clone())
+										},
+									};
+									log_info!(logger, "Failing HTLC with payment_hash {} from {} counterparty commitment tx due to broadcast of revoked counterparty commitment transaction, waiting for confirmation (at height {})", log_bytes!(htlc.payment_hash.0), $commitment_tx, entry.confirmation_threshold());
+									self.onchain_events_waiting_threshold_conf.push(entry);
```

Comment:
I don't have a concrete suggestion for now, was just curious if someone though about it. For sure it can be done as a follow-up if we come with a good code simplification.